src/main/groovy/edu/oregonstate/mist/trailsapi/db/TrailDAO.groovy

```diff
@@ -11,128 +11,160 @@ import org.skife.jdbi.v2.sqlobject.BindBean
+                                    @Bind("woodRide") Boolean woodRide,
+                                    @Bind("skinny") Boolean skinny,
+                                    @Bind("largeJump") Boolean largeJump,
+                                    @Bind("smallJump") Boolean smallJump,
+                                    @Bind("gap") Boolean gap)
 
-	/**************************************************************************************************
-	GET /trails/{trailID}
-	**************************************************************************************************/
-	@SqlQuery("""
-		SELECT
-			TRAILS.ID,
-			TRAILS.NAME,
-			TRAIL_DIFFICULTIES.DIFFICULTY_COLOR,
-			TRAILS.POLYLINE,
-			TRAILS.ZIPCODE,
-			TRAILS.SMALLDROP,
-			TRAILS.LARGEDROP,
-			TRAILS.WOODRIDE,
-			TRAILS.SKINNY,
-			TRAILS.LARGEJUMP,
-			TRAILS.SMALLJUMP,
-			TRAILS.GAP
-		 FROM TRAILS
-		 LEFT JOIN TRAIL_DIFFICULTIES ON TRAILS.DIFFICULTY_ID = TRAIL_DIFFICULTIES.DIFFICULTY_ID
-		 WHERE TRAILS.ID = :id
-	""")
-	Trail getTrailByID(@Bind("id") Integer id)
+        /**************************************************************************************************
+        GET /trails/{trailID}
+        **************************************************************************************************/
+        @SqlQuery("""
+        SELECT
+            TRAILS.ID,
+            TRAILS.NAME,
+            TRAIL_DIFFICULTIES.DIFFICULTY_COLOR,
+            TRAILS.POLYLINE,
+            TRAILS.ZIPCODE,
+            TRAILS.SMALLDROP,
+            TRAILS.LARGEDROP,
+            TRAILS.WOODRIDE,
+            TRAILS.SKINNY,
+            TRAILS.LARGEJUMP,
+            TRAILS.SMALLJUMP,
+            TRAILS.GAP
+         FROM TRAILS
+         LEFT JOIN TRAIL_DIFFICULTIES ON TRAILS.DIFFICULTY_ID = TRAIL_DIFFICULTIES.DIFFICULTY_ID
+         WHERE TRAILS.ID = :id
+    """)
+    Trail getTrailByID(@Bind("id") Integer id)
+
+    /**************************************************************************************************
+    PUT /trails/{trailID}
+    Updates values if binded value is not null
+    **************************************************************************************************/
+    @SqlUpdate("""
+        UPDATE TRAILS
+        SET
+            NAME = COALESCE(:name, NAME),
+            DIFFICULTY_ID = COALESCE((SELECT DIFFICULTY_ID FROM TRAIL_DIFFICULTIES
+            WHERE DIFFICULTY_COLOR = :difficulty), DIFFICULTY_ID),
+            ZIPCODE = COALESCE(:zipCode, ZIPCODE),
+            SMALLDROP = COALESCE(:smallDrop, SMALLDROP),
+            LARGEDROP = COALESCE(:largeDrop, LARGEDROP),
+            WOODRIDE = COALESCE(:woodRide, WOODRIDE),
+            SKINNY = COALESCE(:skinny, SKINNY),
+            LARGEJUMP = COALESCE(:largeJump, LARGEJUMP),
+            SMALLJUMP = COALESCE(:smallJump, SMALLJUMP),
+            GAP = COALESCE(:gap, GAP)
+        WHERE ID = :id
+    """)
+    void updateTrail(@Bind("id") Integer id,
```

Comment:
You used a `@BindBean` in your `POST` query, could you use it here too?

Reply:
BindBean worked on POST only because I assigned all of the Booleans default values of false, which avoid the null Boolean binding bug. I didn't want to do this on PUT because the user may only update one field of the trail and leave all of the Boolean fields null. In hindsight, it probably makes more sense to ditch BindBean in POST as well, because not providing information of say, a small drop does not mean that there isn't a small drop.